common/src/main/java/com/google/auto/common/AnnotationValues.java

```diff
@@ -348,7 +353,7 @@ public static char getChar(AnnotationValue value) {
 
     @Override
     public ImmutableList<T> defaultAction(Object o, Void unused) {
-      throw new IllegalStateException("Expected an array, got instead: " + o);
+      throw new IllegalArgumentException("Expected an array, got instead: " + o);
```

Comment:
This seems correct. Technically, it is an incompatible change, though I think it's unlikely it would break production code. (It might conceivably break tests.) I'm running it past all of Google's internal tests just to check.

Reply:
Tests all pass. We're good to go. Thanks!